Lib/email/_header_value_parser.py

```diff
@@ -1125,7 +1126,7 @@ def get_unstructured(value):
         # Split in the middle of an atom if there is a rfc2047 encoded word
         # which does not have WSP on both sides. The defect will be registered
         # the next time through the loop.
-        if rfc2047_matcher.search(tok):
+        if not encoded_format_invalid and rfc2047_matcher.search(tok):
```

Comment:
This is un-necessary since we don't have a test case where the encoded_word is invalid, but it still would match `rfc2047_matcher.search(tok)`. 

See my more detailed comment on bpo: https://bugs.python.org/msg349847

Reply:
In the test case I just added ('=?utf-8?q?=somevalue?='), this part of code is needed.

Reply:
I was thinking about this and you have correctly found a test case where encoded-word in invalid but still match the regex. 
That is great!

Reply:
Can you also expand the comment above to mention that in use cases where regex matches and encoded-word parsing fails, we go into infinite loop.

Reply:
Done.